pandas/tools/tests/test_util.py

```diff
@@ -291,6 +291,42 @@ def test_non_hashable(self):
         with self.assertRaisesRegexp(TypeError, "Invalid object type"):
             pd.to_numeric(s)
 
+    def test_compact_ints(self):
+        from pandas.parser import na_values
+
+        s = ['1', 2, 3]
+
+        res = pd.to_numeric(s)
+        expected = np.array([1, 2, 3], dtype=np.int64)
+        tm.assert_numpy_array_equal(res, expected)
+
+        res = pd.to_numeric(s, compact_ints=True)
+        expected = np.array([1, 2, 3], dtype=np.int8)
+        tm.assert_numpy_array_equal(res, expected)
+
+        res = pd.to_numeric(s, compact_ints=True, use_unsigned=True)
+        expected = np.array([1, 2, 3], dtype=np.uint8)
+        tm.assert_numpy_array_equal(res, expected)
+
+        # ignore use_unsigned (negative number)
+        s = ['-1', 2, 3]
+        res = pd.to_numeric(s, compact_ints=True, use_unsigned=True)
+        expected = np.array([-1, 2, 3], dtype=np.int8)
+        tm.assert_numpy_array_equal(res, expected)
+
+        # ignore compact_ints (float element)
+        s = ['-1.0', 2, 3]
+        res = pd.to_numeric(s, compact_ints=True)
+        expected = np.array([-1, 2, 3], dtype=np.float64)
+        tm.assert_numpy_array_equal(res, expected)
+
+        # compact int64 "NaN" value
+        int8_na = na_values[np.int8]
+        int64_na = na_values[np.int64]
+        s = [int64_na, 2, 3]
```

Comment:
I know the "NaN" like values is the current `lib.downcast_int64` behaviour, but I am not sure we should put that in user-facing code? 
Because that means you are actually changing the content (and they are currently not used as NaNs)